### PR TITLE
Better Exceptions handling

### PR DESCRIPTION
**Pull request by HavokInspiration**
## A bit of context

I created a Helper to manage menus in an admin layout for a project I'm working on. This helper has a method `display()` that takes an argument which a key that can be configured by the dev. If the key does not match anything configured, the method throws a custom exception.
The problem is that Twig catches any exception and throws is own with a message saying "Hey, an exception was thrown" silencing the exception message I set up in my helper.

## The change

This change makes sure that if an exception is thrown during rendering, and that this exception has a reference to a previous exception (which is the case here, the exception thrown in the helper is the previous one reference in the Twig exception caught in the `TwigView::_render()` method), it should throw the previous exception and not the Twig one.
If no exception is referenced, it just throws the exception that just has been caught.

### CODE REVIEW

**Comment by WyriHaximus** [comment]
One job failed, I've restarted it to give it another go :smile: . Will look into the changes later tonight, thanks :+1:

File: src/View/TwigView.php [hunk 1]
```diff
@@ -179,10 +180,11 @@ protected function generateHelperList()
      * @param string $viewFile Template file.
      * @param array  $data     Data that can be used by the template.
      *
+     * @throws \Exception
      * @return string
      */
     // @codingStandardsIgnoreStart
-    protected function _render($viewFile, $data = array())
+    public function _render($viewFile, $data = array())
```
Comment (HavokInspiration) [inline src/View/TwigView.php]: Seems I changed the visibility of the method, which I should not.
Will fix it as soon as I can.

Comment (WyriHaximus) [inline src/View/TwigView.php]: :+1:

**Comment by HavokInspiration** [comment]
I changed the method visibility back to its previous state. Don't know why I changed it in the first place though 😄

**Comment by WyriHaximus** [comment]
That one fail is interestingly persistent: https://travis-ci.org/WyriHaximus/TwigView/jobs/152966123#L356

**Comment by HavokInspiration** [comment]
Would be worth to launch the test suite on Travis against master to check if this is because of my changes.

**Comment by WyriHaximus** [comment]
Lets see: https://travis-ci.org/WyriHaximus/TwigView/builds/152973371

**Comment by WyriHaximus** [comment]
Yup also on master: https://travis-ci.org/WyriHaximus/TwigView/jobs/152973385#L351

**Comment by HavokInspiration** [comment]
Interesting. Must be a problem on Travis related to the nightly build of PHP 7.

**Comment by WyriHaximus** [comment]
Yeah it seems we can savely ignore it for this PR :smile:

**Comment by WyriHaximus** [comment]
LGTM :shipit:

**Pull request merged**